From 5c9223af57eddc847f67946f0fc36fda134383b6 Mon Sep 17 00:00:00 2001
From: Dechen Chai <quic_dchai@quicinc.com>
Date: Thu, 1 Sep 2022 12:01:17 +0800
Subject: [PATCH] gstrosnode-ok.patch

Change-Id: I5a343efd0a854391c1ddae0884b49f73d14ffba2

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 487e3b76d..5dfffee84 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -26,6 +26,7 @@ set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-missing-field-initializers")
 if (NOT GNU)
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-c99-designator")
 endif ()
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-c99-designator")
 
 if (ENABLE_LIBLOG)
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DENABLE_LIBLOG")
@@ -90,6 +91,8 @@ else ()
 endif ()
 
 set(ROS-COMMON-LIBS
+  rcpputils
+  rosidl_runtime_c
   rclcpp
   rcutils
   yaml
