From b40258d5f846a1a49e2cbf05bf45c4857ea5bf5c Mon Sep 17 00:00:00 2001
From: chalia <quic_chalia@quicinc.com>
Date: Mon, 24 Jan 2022 15:33:29 +0800
Subject: [PATCH] fix_db_mismatch

---
 pseudo.c        | 3 ---
 pseudo_client.c | 6 ------
 2 files changed, 9 deletions(-)

diff --git a/pseudo.c b/pseudo.c
index 528fe1b..4bc5768 100644
--- a/pseudo.c
+++ b/pseudo.c
@@ -703,8 +703,6 @@ pseudo_op(pseudo_msg_t *msg, const char *program, const char *tag, char **respon
 						path_by_ino ? path_by_ino : "no path",
 						msg->path);
 					found_ino = 0;
-					msg->result = RESULT_ABORT;
-					goto op_exit;
 				}
 			}
 		} else {
@@ -1041,7 +1039,6 @@ pseudo_op(pseudo_msg_t *msg, const char *program, const char *tag, char **respon
 		break;
 	}
 
-op_exit:
 	/* in the case of an exact match, we just used the pointer
 	 * rather than allocating space.
 	 */
diff --git a/pseudo_client.c b/pseudo_client.c
index 579db33..69fd350 100644
--- a/pseudo_client.c
+++ b/pseudo_client.c
@@ -2007,12 +2007,6 @@ pseudo_client_op(pseudo_op_t op, int access, int fd, int dirfd, const char *path
 #endif
 		if (result) {
 			pseudo_debug(PDBGF_OP, "(%d) %s", getpid(), pseudo_res_name(result->result));
-			if (result->result == RESULT_ABORT) {
-				char *local_state_dir = pseudo_get_value("PSEUDO_LOCALSTATEDIR");
-				pseudo_diag("abort()ing pseudo client by server request. See https://wiki.yoctoproject.org/wiki/Pseudo_Abort for more details on this.\n"
-					"Check logfile: %s/%s\n", local_state_dir, PSEUDO_LOGFILE);
-				abort();
-			}
 			if (op == OP_STAT || op == OP_FSTAT) {
 				pseudo_debug(PDBGF_OP, " mode 0%o uid %d:%d",
 					(int) result->mode,
-- 
2.25.1

